coders/xcf.c

```diff
@@ -1086,11 +1157,18 @@ static Image *ReadXCFImage(const ImageInfo *image_info,ExceptionInfo *exception)
       (LocaleNCompare((char *) magick,"gimp xcf",8) != 0))
     ThrowReaderException(CorruptImageError,"ImproperImageHeader");
   (void) memset(&doc_info,0,sizeof(XCFDocInfo));
+  doc_info.version=strtoul(magick+10, NULL, 10);
   doc_info.width=ReadBlobMSBLong(image);
   doc_info.height=ReadBlobMSBLong(image);
   if ((doc_info.width > 262144) || (doc_info.height > 262144))
     ThrowReaderException(CorruptImageError,"ImproperImageHeader");
   doc_info.image_type=ReadBlobMSBLong(image);
+  if (doc_info.version >= 4)
+    precision=ReadBlobMSBLong(image);
```

Comment:
Could you change the `precision` to 150 when `doc_info.version == 4 and precision == 0`?